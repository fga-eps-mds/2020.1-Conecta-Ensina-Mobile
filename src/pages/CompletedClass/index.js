import React, {useState} from 'react';
import Theme from '../../../Theme';
import Background1 from '../../components/Background1';
import {
  ContainerVisualAula,
  ListaVisualAula,
  ContainerButtons,
  ContainerTexto,
  ButtonVerMais,
} from './styles';
import CustomText from '../../components/CustomText';

export default function CompletedClass({navigation}) {
  const [aulas, setAulas] = useState([
    {id: '15', nome: 'ExemploAula01'},
    {id: '25', nome: 'ExemploAula02'},
    {id: '35', nome: 'ExemploAula03'},
    {id: '45', nome: 'ExemploAula04'},
  ]);

  const renderItem = ({item}) => {
    return (
      <ContainerVisualAula>
        <ContainerTexto>
          <CustomText smaller black>
            Horario: 16 - 17h
          </CustomText>
          <CustomText smaller black>
            Conteudo
          </CustomText>
          <CustomText smaller black>
            Modalidade
          </CustomText>
          <CustomText smaller black>
            Distancia
          </CustomText>
          <CustomText smaller black>
            {' '}
          </CustomText>
          <CustomText smaller black>
            Dados do Aluno
          </CustomText>
          <CustomText smaller black>
            {' '}
            - Nome
          </CustomText>
          <CustomText smaller black>
            {' '}
            - Serie
          </CustomText>
          <CustomText smaller black>
            {' '}
            - Endereço
          </CustomText>
          <CustomText smaller black>
            {' '}
            - Observação
          </CustomText>
        </ContainerTexto>
        <ContainerButtons>
          <ButtonVerMais>
            <CustomText smaller white>
              Ver mais
            </CustomText>
          </ButtonVerMais>
        </ContainerButtons>
      </ContainerVisualAula>
    );
  };

  return (
    <Theme>
      <Background1 navigation={navigation} page={'TeacherProfile2'}>
        <ListaVisualAula
          numColumns={2}
          data={aulas}
          keyExtractor={(item) => item.id}
          renderItem={renderItem}
        />
      </Background1>
    </Theme>
  );
}
