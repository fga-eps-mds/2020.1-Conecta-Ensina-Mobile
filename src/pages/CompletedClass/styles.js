import styled from 'styled-components/native';

export const ContainerVisualAula = styled.View`
  background-color: ${(props) => props.theme.colors.cinzaClaro};
  padding: 3%;
  width: 144px;
  height: 223px;
  border-radius: 10px;
  margin-right: 20px;
  margin-bottom: 20px;
`;

export const ListaVisualAula = styled.FlatList`
  padding-bottom: 25%;
  align-self: center;
`;

export const ButtonVerMais = styled.TouchableOpacity`
  background-color: ${(props) => props.theme.colors.fundoAzul};
  width: 103px;
  height: 22.5px;
  border-radius: 5px;
  align-items: center;
  justify-content: center;
`;

export const ContainerButtons = styled.View`
  justify-content: space-around;
  align-self: center;
  flex-direction: row;
`;

export const ContainerTexto = styled.View`
  justify-content: flex-start;
  margin-bottom: 10px;
`;
