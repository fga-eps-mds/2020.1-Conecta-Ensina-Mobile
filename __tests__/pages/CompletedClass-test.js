import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CompletedClass from '../../src/pages/CompletedClass';

describe('Testing CompletedClass page', () => {
  const navigation = {
    navigate: jest.fn(),
  };
  test('Should match snapshot', () => {
    const tree = render(<CompletedClass navigation={navigation} />);

    expect(tree).toMatchSnapshot();
  });
});
