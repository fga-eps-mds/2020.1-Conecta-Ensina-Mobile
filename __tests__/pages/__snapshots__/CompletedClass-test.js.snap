// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CompletedClass page Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 22,
        "flexShrink": 1,
        "marginLeft": 30,
        "opacity": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 1000,
          "height": 58,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginLeft": 40,
          "opacity": 1,
          "width": 58,
        }
      }
      testID="userAvatar"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/user_blue.png",
          }
        }
        style={
          Array [
            Object {
              "height": 38,
              "width": 33,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 40,
            "marginRight": 35,
          },
        ]
      }
    >
      <Text
        medium={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": "15",
            "nome": "ExemploAula01",
          },
          Object {
            "id": "25",
            "nome": "ExemploAula02",
          },
          Object {
            "id": "35",
            "nome": "ExemploAula03",
          },
          Object {
            "id": "45",
            "nome": "ExemploAula04",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": "25%",
            "position": "absolute",
          },
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
